cpp/docs/BENCHMARKING.md

```diff
@@ -1,19 +1,25 @@
 # Unit Benchmarking in libcudf
 
-Unit benchmarks in libcudf are written using [Google Benchmark](https://github.com/google/benchmark).
+Unit benchmarks in libcudf are written using [NVBench](https://github.com/NVIDIA/nvbench).
+While many existing benchmarks are written using
+[Google Benchmark](https://github.com/google/benchmark), new benchmarks should use NVBench.
 
-Google Benchmark provides many options for specifying ranges of parameters to benchmarks to test
-with varying parameters, as well as to control the time unit reported, among other options. Refer to
-other benchmarks in `cpp/benchmarks` to understand the options.
+The NVBench library is similar to Google Benchmark, but has several quality of life improvements
+when doing GPU benchmarking such as displaying the fraction of peak memory bandwidth achieved and
+details about the GPU hardware.
+
+Both NVBench and Google Benchmark provide many options for specifying ranges of parameters to
+benchmark, as well as to control the time unit reported, among other options. Refer to existing
+benchmarks in `cpp/benchmarks` to understand the options.
 
 ## Directory and File Naming
 
 The naming of unit benchmark directories and source files should be consistent with the feature
 being benchmarked. For example, the benchmarks for APIs in `copying.hpp` should live in
-`cudf/cpp/benchmarks/copying`. Each feature (or set of related features) should have its own
+`cpp/benchmarks/copying`. Each feature (or set of related features) should have its own
 benchmark source file named `<feature>_benchmark.cu/cpp`. For example,
-`cudf/cpp/src/copying/scatter.cu` has benchmarks in
-`cudf/cpp/benchmarks/copying/scatter_benchmark.cu`.
+`cpp/src/copying/scatter.cu` has benchmarks in
+`cpp/benchmarks/copying/scatter_benchmark.cu`.
```

Comment:
I've been naming `nvbench` benchmark files with `_nvbench.cu/cpp` suffix and executable with `<feature>_NVBENCH` (instead of `<feature>_BENCH`) to distinguish them from gbench. Not sure it's the right way to go for all `nvbench` benchmarks though.

Reply:
I would prefer no suffix on the source files.  As nvbench becomes the primary benchmark it becomes just noise.

Reply:
@robertmaynard Does "no suffix" mean "remove `_benchmark`" as well?

I find `_benchmark` helpful for finding files by name, but I agree that `_nvbench` is too specific.

Reply:
I recall I decided to do so because of `CMake`:

- `ConfigureBench` to build `_benchmark.cu/cpp`
- `ConfigureNVBench` to build `_nvbench.cu/cpp`

Reply:
> @robertmaynard Does "no suffix" mean "remove `_benchmark`" as well?
> 
> I find `_benchmark` helpful for finding files by name, but I agree that `_nvbench` is too specific.

I was thinking of tool specific suffix. So `benchmark` seems okay to me. The purist in me says that we have `benchmarks` in the path to the file already so no future distinction is required ( we don't use `scatter_header.h` ).


> I recall I decided to do so because of `CMake`:
> 
> * `ConfigureBench` to build `_benchmark.cu/cpp`
> * `ConfigureNVBench` to build `_nvbench.cu/cpp`

`ConfigureBench` and `ConfigureNVBench` are given a set of souce files to compile, and don't care how those files are named. So they would have no problem with `scatter_badly_named.cu` :)

The primary issue is that we have `join_nvbench.cu` and `join_benchmark.cu`, where the nvbench version tests the left_join, full_join, and inner_join algorithms, and google benchmarks tests the semi and anti joins. If we had gone with file names that matched the algorithms ( `anti_join_benchmark.cu`, `left_join_benchmark.cu` ) we wouldn't need to have anything end with `_nvbench`.

Reply:
> ConfigureBench and ConfigureNVBench are given a set of souce files to compile, and don't care how those files are named.

Right. Sorry I was not clear. My point was that I chose to use distinct suffixes mainly for clarity. So people will know at first glance why certain files are built via `ConfigureBench` while others use `ConfigureNVBench`.

Reply:
So do we agree that no change is needed in the guide? (Perhaps some changes are needed to rename `_nvbench.cu/cpp` files.)

Reply:
I prefer to ideally lose suffixes.
Once our benchmarks become large enough to need to split across many files, we might need to add suffixes based on context of that benchmark. At that point, we would prefer to lose _nvbench, _benchmark suffixes because of long filenames.

Reply:
@karthikeyann I can see the argument to remove `_benchmark`. However, that wouldn't align with the current state of the benchmarks. I'd like to keep this PR focused on the change to "use nvbench" and that change can be done at a later point.

Reply:
Just make sure when someone goes through and removes _benchmark from every benchmark source file, they also update the guide to not use the suffix. (If it were me, I would change the example in the guide now...)

Reply:
@harrism I'll merge this PR and do that now. 👍